bootstrap/bootstrap.ps1

```diff
@@ -0,0 +1,107 @@
+param
+(
+    [string]$RepositoryRoot,
+    [string]$ToolsLocalPath,
+    [switch]$Force = $false
+)
+
+$rootToolVersions = Join-Path $RepositoryRoot ".toolversions"
+$bootstrapComplete = Join-Path $ToolsLocalPath "bootstrap.complete"
+
+# if the force switch is specified delete the semaphore file if it exists
+if ($Force -and (Test-Path $bootstrapComplete))
+{
+    del $bootstrapComplete
+}
+
+# if the semaphore file exists and is identical to the specified version then exit
+if ((Test-Path $bootstrapComplete) -and !(Compare-Object (Get-Content $rootToolVersions) (Get-Content $bootstrapComplete)))
```

Comment:
This doesn't take into account just changes to the cli version. I think that is probably OK for now but we need to make sure we can update cli version and the right thing will happen.